zcash_primitives/src/transaction/mod.rs

```diff
@@ -62,6 +68,12 @@ impl Deref for Transaction {
     }
 }
 
+impl DerefMut for Transaction {
```

Comment:
This is risky, because it allows the transaction to be mutated to be invalid - it would be better to implement `From<Transaction> for TransactionData` - this would then allow you to get the TransactionData, which you could then mutate and re-freeze (while invalidating/consuming the original `Transaction`).

Reply:
Cool, OK. I can work around it not being mutable.